components/tabs/tabset.component.ts

```diff
@@ -84,12 +87,41 @@ export class Tabset implements OnInit {
       return;
     }
     // Select a new tab if the tab to be removed is selected and not destroyed
-    if (tab.active && this.tabs.length > 1) {
-      // If this is the last tab, select the previous tab. else, the next tab.
-      let newActiveIndex = index === this.tabs.length - 1 ? index - 1 : index + 1;
+    if (tab.active && this.hasAvailableTabs(index)) {
+      let newActiveIndex = this.getClosestTabIndex(index);
       this.tabs[newActiveIndex].active = true;
     }
 
-    this.tabs.slice(index, 1);
+    tab.removed.emit(tab);
+    this.tabs.splice(index, 1);
+  }
+
+  private getClosestTabIndex (index:number):number {
+    let tabsLength = this.tabs.length;
+    if (tabsLength) {
```

Comment:
Logic was not changed, meaning:
1. check "active", only if True, call "hasAvailableTabs".
2. hasAvailableTabs, at first check tabs length as it was.
3. Additional logic is to check if there is any sense to search for Tab that could be switched to active.
4. And finally only if all negative cases skipped, call "getClosestTabIndex", that will start to search in both directions left/right from current index.

So, previous logic was saved, but enlarged, is it OK ?